echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/jira/JiraNotificationService.java

```diff
@@ -37,36 +40,85 @@
 import org.springframework.stereotype.Component;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import retrofit.RetrofitError;
+import retrofit.client.Response;
 
 @Component
 @ConditionalOnProperty("jira.enabled")
 public class JiraNotificationService implements NotificationService {
   private static final Logger LOGGER = LoggerFactory.getLogger(JiraNotificationService.class);
   private static final int MAX_RETRY = 3;
-  private static final long RETRY_BACKOFF = 3;
+  private static final long RETRY_BACKOFF = 100;
 
   private final JiraService jiraService;
   private final RetrySupport retrySupport;
   private final ObjectMapper mapper;
 
   @Autowired
-  public JiraNotificationService(JiraService jiraService, ObjectMapper objectMapper) {
+  public JiraNotificationService(
+      JiraService jiraService, RetrySupport retrySupport, ObjectMapper objectMapper) {
     this.jiraService = jiraService;
-    this.retrySupport = new RetrySupport();
+    this.retrySupport = retrySupport;
     this.mapper = objectMapper;
   }
 
   @Override
   public boolean supportsType(String type) {
-    return "JIRA".equals(type.toUpperCase());
+    return "JIRA".equalsIgnoreCase(type);
   }
 
   @Override
-  public EchoResponse<CreateJiraIssueResponse> handle(Notification notification) {
+  public EchoResponse handle(Notification notification) {
+    return isTransition(notification) ? transitionIssue(notification) : create(notification);
+  }
+
+  private boolean isTransition(Notification notification) {
+    return notification.getAdditionalContext().get("transitionContext") != null;
+  }
+
+  private EchoResponse.Void transitionIssue(Notification notification) {
+    Map<String, Object> transitionContext =
```

Comment:
Maybe create an actual model object around this and objectMapper.convert() to it?

Makes it easier than all this magical casting?

Reply:
Oh yea for sure.  I'll probably keep the transitionContext bit untyped since I really just want to proxy things through to JIRA, but I'll remove the casting.